src/CSVMsg.chpl

```diff
@@ -0,0 +1,655 @@
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+            }
+            var tmp = glob(filelist[0]);
+            csvLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                  "glob expanded %s to %i files".format(filelist[0], tmp.size));
+            if tmp.size == 0 {
+                var errorMsg = "The wildcarded filename %s either corresponds to files inaccessible to Arkouda or files of an invalid format".format(filelist[0]);
+                csvLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+            }
+            // Glob returns filenames in weird order. Sort for consistency
+            sort(tmp);
+            filedom = tmp.domain;
+            filenames = tmp;
+        } else {
+            // assumes that we are providing 
+            filenames = filelist;
+        }
+
+        var row_cts: [filedom] int;
+        var data_types: list(list(string));
+        var headers: [filedom] bool;
+        var rtnData: list((string, string, string));
+        var fileErrors: list(string);
+        var fileErrorCount:int = 0;
+        var fileErrorMsg:string = "";
+        var validFiles: [filedom] bool = true;
+        for (i, fname) in zip(filedom, filenames) {
+            var hadError = false;
+            try {
+                var dtypes: list(string);
+                (row_cts[i], headers[i], dtypes) = get_info(fname, dsetlist, col_delim);
+                data_types.append(dtypes);
+            } catch e: FileNotFoundError {
+                fileErrorMsg = "File %s not found".format(fname);
+                csvLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                hadError = true;
+                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+            } catch e: PermissionError {
+                fileErrorMsg = "Permission error %s opening %s".format(e.message(),fname);
+                csvLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                hadError = true;
+                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
+            } catch e: DatasetNotFoundError {
+                fileErrorMsg = "1 or more Datasets not found in file %s".format(fname);
+                csvLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                hadError = true;
+                if !allowErrors { return new MsgTuple(e.message(), MsgType.ERROR); }
+            } catch e: NotHDF5FileError {
+                fileErrorMsg = "The file %s is not an HDF5 file: %s".format(fname,e.message());
+                csvLogger.error(getModuleName(),getRoutineName(),getLineNumber(),fileErrorMsg);
+                hadError = true;
+                if !allowErrors { return new MsgTuple(fileErrorMsg, MsgType.ERROR); }
```

Comment:
we probably don't need this error check

Reply:
That is correct. Forgot to remove when I stole the block of code from HDF5.